libindy-crypto/src/cl/verifier.rs

```diff
@@ -0,0 +1,367 @@
+use bn::BigNumber;
+use cl::*;
+use cl::constants::{LARGE_E_START, ITERATION, LARGE_NONCE};
+use cl::helpers::*;
+use errors::IndyCryptoError;
+
+use std::collections::{HashMap, HashSet};
+
+/// Party that wants to check that prover has some credentials provided by issuer.
+pub struct Verifier {}
+
+impl Verifier {
+    /// Creates and returns sub proof request entity builder.
+    ///
+    /// The purpose of sub proof request builder is building of sub proof request entity that
+    /// represents requested attributes and predicates.
+    ///
+    /// # Example
+    /// ```
+    /// use indy_crypto::cl::verifier::Verifier;
+    /// use indy_crypto::cl::Predicate;
+    /// let mut sub_proof_request_builder = Verifier::new_sub_proof_request().unwrap();
+    /// sub_proof_request_builder.add_revealed_attr("name").unwrap();
+    ///
+    /// let predicate = Predicate::new("age", "GE", 18).unwrap();
+    /// sub_proof_request_builder.add_predicate(&predicate).unwrap();
+    ///
+    /// let _sub_proof_request = sub_proof_request_builder.finalize().unwrap();
+    /// ```
+    pub fn new_sub_proof_request() -> Result<SubProofRequestBuilder, IndyCryptoError> {
```

Comment:
Please describe what sub-proof request is (this is part of proof request related to a particular schema-key)